### PR TITLE
Safely remove all usage of `!![`

### PR DESCRIPTION
**Pull request by bmarty**
Especially to stabilize crypto code

Tested:
- Send e2e message
- Receive e2e message
- Incoming device verification

### CODE REVIEW

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/model/create/CreateRoomParams.kt [hunk 1]
```diff
@@ -210,13 +210,11 @@ class CreateRoomParams {
      * @return the first invited user id
      */
     fun getFirstInvitedUserId(): String? {
-        if (0 != getInviteCount()) {
-            return invitedUserIds!![0]
+        if (getInviteCount() > 0) {
```
Comment (BillCarsonFr) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/model/create/CreateRoomParams.kt]: Same as: 
return invitedUserIds?.firstOrNull() ?: invite3pids?.firstOrNull()?.address

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/keysbackup/KeysBackup.kt [hunk 2]
```diff
@@ -388,8 +388,8 @@ internal class KeysBackup @Inject constructor(
             return keysBackupVersionTrust
         }
 
-        val mySigs: Map<String, *> = authData.signatures!![myUserId] as Map<String, *>
-        if (mySigs.isEmpty()) {
+        val mySigs = authData.signatures?.get(myUserId)
+        if (mySigs == null || mySigs.isEmpty()) {
```
Comment (BillCarsonFr) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/keysbackup/KeysBackup.kt]: .isNullOrEmpty()

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/keysbackup/KeysBackup.kt [hunk 3]
```diff
@@ -402,9 +402,8 @@ internal class KeysBackup @Inject constructor(
                 deviceId = components[1]
             }
 
-            var device: MXDeviceInfo? = null
             if (deviceId != null) {
-                device = cryptoStore.getUserDevice(deviceId, myUserId)
+                var device = cryptoStore.getUserDevice(deviceId, myUserId)
```
Comment (BillCarsonFr) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/keysbackup/KeysBackup.kt]: val?

there is also a remaining !! at line 414 device.fingerprint()!!

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/model/MXDeviceInfo.kt [hunk 4]
```diff
@@ -107,30 +106,25 @@ data class MXDeviceInfo(
...
 
     /**
      * @return the display name
      */
     fun displayName(): String? {
-        return if (null != unsigned) {
-            unsigned!!["device_display_name"] as String?
-        } else null
-
+        return unsigned?.get("device_display_name") as String?
```
Comment (BillCarsonFr) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/model/MXDeviceInfo.kt]: as? String 
or ?.toString()

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/model/create/CreateRoomParams.kt (hunk 1 above)
Comment (bmarty) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/model/create/CreateRoomParams.kt]: done

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/keysbackup/KeysBackup.kt (hunk 2 above)
Comment (bmarty) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/keysbackup/KeysBackup.kt]: done

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/keysbackup/KeysBackup.kt (hunk 3 above)
Comment (bmarty) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/keysbackup/KeysBackup.kt]: done

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/model/MXDeviceInfo.kt (hunk 4 above)
Comment (bmarty) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/model/MXDeviceInfo.kt]: done

**Pull request merged**